docs/source/index.rst

```diff
@@ -1,5 +1,18 @@
+.. toctree::
+    :hidden:
+    :maxdepth: 1
+
+    install
+    getting_started
+    main_concepts
+    user_guide
+    examples
+    api_reference
+    changelog
```

Comment:
let's rename changelog to "release notes".  In general, let's try to match the EvalML navbar scheme